exchanges/binanceus/binanceus_wrapper.go

```diff
@@ -0,0 +1,1005 @@
+	// An endpoint like /api/v3/allOrders does not exist in the binance us
+	// so This end point is left Un Implemented
+	return nil, common.ErrFunctionNotSupported
+}
+
+// GetFeeByType returns an estimate of fee based on the type of transaction
+func (bi *Binanceus) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	if feeBuilder == nil {
+		return 0, fmt.Errorf("%T %w", feeBuilder, common.ErrNilPointer)
+	}
+	if (!bi.AreCredentialsValid(ctx) || bi.SkipAuthCheck) && // Todo check connection status
+		feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		feeBuilder.FeeType = exchange.OfflineTradeFee
+	}
+	return bi.GetFee(ctx, feeBuilder)
+}
+
+// ValidateCredentials validates current credentials used for wrapper
+func (bi *Binanceus) ValidateCredentials(ctx context.Context, assetType asset.Item) error {
+	_, err := bi.UpdateAccountInfo(ctx, assetType)
+	return bi.CheckTransientError(err)
+}
+
+// GetHistoricCandles returns candles between a time period for a set time interval
+func (bi *Binanceus) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if err := bi.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+	if kline.TotalCandlesPerInterval(start, end, interval) > float64(bi.Features.Enabled.Kline.ResultLimit) {
+		return kline.Item{}, errors.New(kline.ErrRequestExceedsExchangeLimits)
+	}
+	req := KlinesRequestParams{
+		Interval:  bi.GetIntervalEnum(interval),
+		Symbol:    pair,
+		StartTime: start,
+		EndTime:   end,
+		Limit:     int(bi.Features.Enabled.Kline.ResultLimit),
+	}
+	ret := kline.Item{
+		Exchange: bi.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+
+	candles, err := bi.GetSpotKline(ctx, &req)
+	if err != nil {
+		return kline.Item{}, err
+	}
+	for x := range candles {
+		ret.Candles = append(ret.Candles, kline.Candle{
+			Time:   candles[x].OpenTime,
+			Open:   candles[x].Open,
+			High:   candles[x].High,
+			Low:    candles[x].Low,
+			Close:  candles[x].Close,
+			Volume: candles[x].Volume,
+		})
+	}
+	ret.SortCandlesByTimestamp(false)
+	return ret, nil
+}
+
+// GetHistoricCandlesExtended returns candles between a time period for a set time interval
+func (bi *Binanceus) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if err := bi.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+	ret := kline.Item{
+		Exchange: bi.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+	dates, err := kline.CalculateCandleDateRanges(start, end, interval, bi.Features.Enabled.Kline.ResultLimit)
+	if err != nil {
+		return kline.Item{}, err
+	}
+	var candles []CandleStick
+	for x := range dates.Ranges {
+		req := KlinesRequestParams{
+			Interval:  bi.FormatExchangeKlineInterval(interval),
```

Comment:
This is causing issues in parsing to what Binance wants.

```suggestion
			Interval:  bi.GetIntervalEnum(interval),
```